Rules/CommonScripts/MapVotesCommon.as

```diff
@@ -596,94 +597,60 @@ void CreateMapTexture(string shortname, string filename)
-					{
-						editcol = colors::minimap_solid_edge;
-					}
-					else if (tile_u == ColTileType::Gold || tile_d == ColTileType::Gold || tile_l == ColTileType::Gold || tile_r == ColTileType::Gold)
-					{
-						editcol = colors::minimap_gold_edge;
-					}
-				}
-				else if ( tile == ColTileType::Backwall && PixelCol != SColor(map_colors::tile_grass) )
-				{
-					//Background
-					editcol = colors::minimap_back;
-
-					//Edge
-					if ( tile_u == ColTileType::Sky || tile_d == ColTileType::Sky || tile_l == ColTileType::Sky || tile_r == ColTileType::Sky ||
-						 tile_u == ColTileType::Sky_Water || tile_d == ColTileType::Sky_Water || tile_l == ColTileType::Sky_Water || tile_r == ColTileType::Sky_Water )
-					{
-						editcol = colors::minimap_back_edge;
-					}
-				}
-				else if (tile == ColTileType::Gold)
-				{
-					//Gold
-					editcol = colors::minimap_gold;
-
-					//Edge
-					if ( (tile_u != ColTileType::Solid && tile_u != ColTileType::Gold) || (tile_d != ColTileType::Solid && tile_d != ColTileType::Gold) ||
-						 (tile_l != ColTileType::Solid && tile_l != ColTileType::Gold) || (tile_r != ColTileType::Solid && tile_r != ColTileType::Gold) )
-					{
-						editcol = colors::minimap_gold_exposed;
-					}
-				}
-				else if ( tile == ColTileType::Backwall_Water )
-				{
-					//Background
-					editcol = colors::interpolated_water_backwall;
-
-					//Edge
-					if ( tile_u == ColTileType::Sky || tile_d == ColTileType::Sky || tile_l == ColTileType::Sky || tile_r == ColTileType::Sky ||
-						 tile_u == ColTileType::Sky_Water || tile_d == ColTileType::Sky_Water || tile_l == ColTileType::Sky_Water || tile_r == ColTileType::Sky_Water )
-					{
-						editcol = colors::interpolated_water_backwall_edge;
-					}
-				}
-				else if (tile == ColTileType::Sky_Water)
-				{   //Sky
-					editcol = colors::interpolated_water_sky;
-				}
-				else
-				{   //Sky
-					editcol = colors::minimap_open;
+				
+				show_gold = true;
+				
+				///Colors
+				const SColor color_minimap_open         (0xffA5BDC8);
+				const SColor color_minimap_ground       (0xff844715);
+				const SColor color_minimap_back         (0xff3B1406);
+				const SColor color_minimap_stone        (0xff8B6849);
+				const SColor color_minimap_thickstone   (0xff42484B);
+				const SColor color_minimap_gold         (0xffFEA53D);
+				const SColor color_minimap_bedrock      (0xff2D342D);
+				const SColor color_minimap_wood         (0xffC48715);
+				const SColor color_minimap_castle       (0xff637160);
+				
+				const SColor color_minimap_castle_back  (0xff313412);
+				const SColor color_minimap_wood_back    (0xff552A11);
+				
+				const SColor color_minimap_water        (0xff2cafde);
+				const SColor color_minimap_fire         (0xffd5543f);
+				
+				
+				if        (PixelCol == SColor(map_colors::tile_ground))           { editcol = color_minimap_ground;
```

Comment:
use a switch instead

Reply:
when i try doing
```
                switch(PixelCol)
                {
                    case SColor(map_colors::tile_ground):            editcol = color_minimap_ground;
                    case SColor(map_colors::tile_ground_back):       editcol = color_minimap_back;
                    case SColor(map_colors::tile_stone):             editcol = color_minimap_stone;
```
game says
[15:35:53] Script: Switch expressions must be integral numbers
[15:35:53]         [line 623 char 12] 				switch(PixelCol)
[15:35:53]         in file '../Mods/MapR/Base/Rules/CommonScripts/MapVotesCommon.as'

Reply:
fixed